### PR TITLE
Testing/update cypress tests

### LINKED ISSUE #79: PUT and POST cypress intercepts

### PR DESCRIPTION
**Pull request by KatherineBlaine**
### Files Changed:

- Home.js
- Home.css
- Header.js

All cypress specs were updated!

### What Has Changed: 
- Updated the Header component so it wont render on the home page
- Updated styling for the What is Beam dropdown
- Finally got the POST intercept figured out. Added that at the end of the Activities spec. 
- Added the PUT intercept and tested updating journal entries
- Did some minor cleanup/updates for failing tests as a result of component changes!
- Added a postedEntries fixture to test posting data to the DOM.

### Notes: (includes questions, ideas for the future, anything you want to call out)

- I believe all that is left for testing is to test the error handling that Chrissy and Joe are finishing up for network request errors, that should cover everything! Lmk if you think of another user path that should be tested, happy to add more!

### Closes: (any issues that it closes)

closes #79

### CODE REVIEW

File: client/cypress/e2e/activities-spec.cy.js [hunk 1]
```diff
@@ -16,22 +16,35 @@ describe('Activities page', () => {
...
+    cy.get('.activity-card > :nth-child(1)').should('be.visible').click()
+    cy.get('h2').should('be.visible').contains('Take a long shower or bath, and change into clothes that make you feel comfortable and confident!')
   })
 
-  it('Should display a button that selects the current activity and brings the user from the activities page to the log page', () => {
+  it('Should display a button that selects the current activity and brings the user from the activities page to the log page. The log page should include the original data along with the posted data.', () => {
+    cy.intercept('GET', 'http://localhost:3001/api/v1/entries', {
+      statusCode: 200,
+      fixture: 'postEntries.json'
+    })
```
Comment (jfogiato) [inline client/cypress/e2e/activities-spec.cy.js:40]: HECK YES! Great work KB! Stoked to see this all come together!

File: client/cypress/e2e/entry-spec.cy.js [hunk 2]
```diff
@@ -23,11 +26,25 @@ describe('template spec', () => {
...
+    cy.get('a.uni-btn').click()
+  })
+
+  it('Should allow the use to cancel a journal entry edit and return the entry entry to its previous value', () => {
+    cy.get('button.uni-btn').click()
+    cy.get('.journal-text-edit').should('be.visible').type(' But after I went on a walk I felt much more calm!')
+    cy.get(':nth-child(4) > :nth-child(1)').click()
+    cy.get('.journal-content').should('be.visible').contains('Today I am feeling anxious because I have a big test coming up')
+  })
+
```
Comment (jfogiato) [inline client/cypress/e2e/entry-spec.cy.js:43]: I just love how thorough your testing is. You truly thought of everything. Thank you!!

File: client/cypress/e2e/entry-spec.cy.js [hunk 3]
```diff
@@ -23,11 +26,25 @@ describe('template spec', () => {
...
   it('Should display a button that brings the user from the entry page back to the log page', () => {
-    cy.get('.uni-btn').click()
+    cy.get('a.uni-btn').click()
```
Comment (jfogiato) [inline client/cypress/e2e/entry-spec.cy.js:45]: a

File: client/cypress/e2e/home-spec.cy.js [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
-  })
-
-  it('Should display a final subheading message', () => {
-    cy.get('.home > :nth-child(4)').should('be.visible').contains('We hope this process will help reveal useful patterns over time.')
+  it('Should display the home blurb when the user selects the dropdown', () => {
+    cy.get('.whats-beam').click()
+    cy.get('h2').should('be.visible').contains('Beam is an interactive self-care feeling tracking app.')
+    cy.get('.blurb').should('be.visible').contains('We created a flow intended to help you work through any kind of feeling or emotion you might be experiencing. You\'ll be asked to choose a feeling and reflect on it. Add anything about your experience that might help you avoid or recreate the feeling in the future (depending on the feeling). After your reflection, you\'ll see an offering of a few activities that might help. Your entries will be available to you in the Feelings Archive for future reflection. We hope this process will help reveal useful patterns over time.')
```
Comment (jfogiato) [inline client/cypress/e2e/home-spec.cy.js:26]: Thank you for your work (1) restyling the homepage dropdown, and (2) subsequently rewriting all the tests for it lol

File: client/src/Components/Header/Header.js [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
-  );
+    <Switch>
+      <Route exact path='/'></Route>
+      <Route path='*'>
+        <header>
+            <p className='logo'>BEAM</p>  
+        </header>
+      </Route>
+    </Switch>
+  )
```
Comment (jfogiato) [inline client/src/Components/Header/Header.js:15]: UNYEET

File: client/src/Components/Home/Home.js [hunk 6]
```diff
@@ -8,16 +8,18 @@ const Home = () => {
   return (
     <section className='home'>
       <h1 className='logo-home'>BEAM</h1>
-      <p className='uni-btn' onClick={() => setBlurb(!blurb)}>What is Beam?</p>
+        <Link to='/how-are-you-feeling' className='uni-btn'>Let&apos;s dig in.</Link>
+        <div className='learn-more'>
+      <p className='whats-beam' onClick={() => setBlurb(!blurb)}>What is Beam?{!blurb ? <i className="fa-solid fa-angle-down"></i> : <i className="fa-solid fa-angle-up"></i>}</p>
+        </div>
```
Comment (jfogiato) [inline client/src/Components/Home/Home.js:14]: Loving this new styling so much.

**Review by jfogiato — APPROVED** [review]
Once again, thank you for the massive work you did with testing! Everything looks really really good. Appreciate all your effort KB!

File: client/cypress/e2e/entry-spec.cy.js (hunk 2 above)
Comment (espressoGoddess) [reply client/cypress/e2e/entry-spec.cy.js:43]: agreed!

File: client/cypress/e2e/home-spec.cy.js [hunk 7]
```diff
@@ -1,26 +1,29 @@
...
   })
 
   it('Should display the site logo', () => {
     cy.get('.logo-home').should('be.visible').contains('BEAM')
   })
 
-  it('Should display a heading for the site info blurb' , () => {
-    cy.get('.home > :nth-child(2)').should('be.visible').contains('Beam is an interactive self-care feeling tracking app.')
+  it('Should display a dropdown for the site info blurb' , () => {
+    cy.get('.whats-beam').should('be.visible').contains('What is Beam?')
```
Comment (espressoGoddess) [inline client/cypress/e2e/home-spec.cy.js:20]: i love how you use the` be.visible`, i hadn't used that before! love how simple it makes this

File: client/cypress/e2e/prompt-spec.cy.js [hunk 8]
```diff
@@ -22,11 +22,21 @@ describe('Journal prompt spec', () => {
...
-    cy.get('.uni-btn').should('be.visible').contains('→').click()
+    cy.get('#journal-entry').type('Im feeling anxious because I have a big project due this weekend.')
+    cy.get('.uni-btn').should('be.visible').click()
     cy.url().should('eq', 'http://localhost:3000/what-should-you-do/Anxious')
   })
+
+  it('Should only allow the user to continue to the next page if they have added a journal entry, and display an error message if the entry box is left blank. The user should then be able to complete an entry and successfully continue to the next page.', () => {
+    cy.get('.uni-btn').click()
+    cy.get('.prompt-container').should('be.visible')
+    cy.get('[style="text-align: center;"]').should('be.visible').contains('Please tell us a bit more about how you\'re feeling today - you can always edit it later. 🤍')
```
Comment (espressoGoddess) [inline client/cypress/e2e/prompt-spec.cy.js:36]: ++ sad path!

File: client/src/Components/Home/Home.js (hunk 6 above)
Comment (espressoGoddess) [reply client/src/Components/Home/Home.js:14]: ++

**Review by espressoGoddess — APPROVED** [review]

**Review by chrissycooper — APPROVED** [review]
Excellent work KB I really appreciate it!

**Pull request merged**